### ISSUE TITLE
Precision-related test failures on i686 Linux, release-0.6

### ISSUE THREAD

**Issue by ararslan** [JuliaLang/julia#24959]
Observed on Travis and seems only to happen on i686.
```
linalg/dense: Test Failed
  Expression: norm(reshape(A, length(A)), p) == vecnorm(A, p)
   Evaluated: 8.422609468915075 == 8.422609468915073
Stacktrace:
 [1] record(::Base.Test.DefaultTestSet, ::Base.Test.Fail) at ./test.jl:568
 [2] (::##40#46)() at /tmp/julia/share/julia/test/runtests.jl:160
 [3] cd(::##40#46, ::String) at ./file.jl:70
 [4] include_from_node1(::String) at ./loading.jl:576
 [5] include(::String) at ./sysimg.jl:14
 [6] process_options(::Base.JLOptions) at ./client.jl:305
 [7] _start() at ./client.jl:371
linalg/dense: Test Failed
  Expression: norm(reshape(A, length(A)), p) == vecnorm(A, p)
   Evaluated: 44.706942422184675 == 44.70694242218468
Stacktrace:
 [1] record(::Base.Test.DefaultTestSet, ::Base.Test.Fail) at ./test.jl:568
 [2] (::##40#46)() at /tmp/julia/share/julia/test/runtests.jl:160
 [3] cd(::##40#46, ::String) at ./file.jl:70
 [4] include_from_node1(::String) at ./loading.jl:576
 [5] include(::String) at ./sysimg.jl:14
 [6] process_options(::Base.JLOptions) at ./client.jl:305
 [7] _start() at ./client.jl:371
```
(and a lot more)

**Comment by ararslan** [comment]
I'm actually seeing this as far back as 0.6.0.

**Comment by ararslan** [comment]
Seems to work with the 32-bit Linux binary for 0.5.2.

**Comment by ararslan** [comment]
Wow, unexpected plot twist: Building 0.6.0 and 0.6.1 from source I see the failure, but I can't reproduce the failure using the official 32-bit binaries.

**Comment by ararslan** [comment]
Problem with configuration used on Travis. Fixed by #24976.
